internal/cli/command_factory.go

```diff
@@ -153,7 +142,8 @@ func (factory *CommandFactory) Close() {
 // Run executes the command
 func (factory *CommandFactory) Run(cmd *cobra.Command) {
 	if err := cmd.Execute(); err != nil {
-		if _, ok := err.(errSupressUsage); !ok {
+		cause, ok := errors.Unwrap(err).(UsagePrinter)
+		if !ok || cause.PrintUsage() {
```

Comment:
the way we handle PrintUsage is a little confusing as returning false seems like an antipattern. Since false is the zero value of a bool, would it make more sense to flip this to `DisablePrintUsage`.